packages/workbox-build/src/entry-points/generate-sw.js

```diff
@@ -47,35 +50,29 @@ async function generateSW(input) {
   const options = validate(input, generateSWSchema);
 
   const destDirectory = path.dirname(options.swDest);
-  // This will cause the dev build of WorkboxSW (either from the CDN or locally
-  // copying it) to be used if NODE_ENV starts with 'dev'.
-  const buildType = (process.env.NODE_ENV &&
-    process.env.NODE_ENV.startsWith('dev')) ? 'dev' : 'prod';
 
   if (options.importWorkboxFromCDN) {
-    const cdnUrl = cdnUtils.getModuleUrl('workbox-sw', buildType);
+    // TODO: Update for https://github.com/GoogleChrome/workbox/issues/969.
+    const cdnUrl = cdnUtils.getModuleUrl('workbox-sw', 'prod');
     // importScripts may or may not already be an array containing other URLs.
-    options.importScripts = (options.importScripts || []).concat(cdnUrl);
+    // Either way, list cdnUrl first.
+    options.importScripts = [cdnUrl].concat(options.importScripts || []);
   } else {
-    // If we're not importing the Workbox script from the CDN, copy it over.
-    const pathToWorkboxSWFile = await copyWorkboxSW(destDirectory, buildType);
-
-    // If we're writing our SW file to build/sw.js, the workbox-sw file will be
-    // build/workbox-sw.js. So the sw.js file should import workboxSW.***.js
-    // (i.e. not include build/).
-    const pathToWorkboxSWFileRelativeToDest = path.relative(destDirectory,
-      pathToWorkboxSWFile);
+    // If we're not importing the Workbox scripts from the CDN, then copy
+    // over the dev + prod version of all of the core libraries.
+    const workboxDirectoryName = await copyWorkboxLibraries(destDirectory);
 
-    // Add a few extra ignore rules to whatever might be specified to avoid
-    // picking up the generated service worker or the workbox-sw files.
+    // The Workbox library files should not be precached, since they're cached
+    // automatically by virtue of being used with importScripts().
     options.globIgnores = [
-      path.basename(pathToWorkboxSWFileRelativeToDest),
-      `${path.basename(pathToWorkboxSWFileRelativeToDest)}.map`,
-    ].map((file) => `**/${file}`).concat(options.globIgnores);
+      `**/${workboxDirectoryName}/*.js*`,
```

Comment:
Could you make sure there are tests that cover the workbox directory being at the root of the glob directory as well as being nested.

Just unsure if glob patterns will match if there is no nesting with this pattern. (Only mentioning because of the number of issues people raised after my first attempt at workbox-build)

Reply:
Sure, I've added an end-to-end test that specifically exercises the `globIgnores` effect for both top- and sub-directory matches.